spacy/kb/kb.pyx

```diff
@@ -32,27 +32,14 @@ cdef class KnowledgeBase:
         self.entity_vector_length = entity_vector_length
         self.mem = Pool()
 
-    def get_candidates_batch(
-            self, mentions: SpanGroup
-    ) -> Iterable[Iterable[Candidate]]:
+    def get_candidates(self, mentions: Iterator[SpanGroup]) -> Iterator[Iterable[Iterable[Candidate]]]:
```

Comment:
Three levels of iterator indirection seems a bit confusing to me. Would a concrete (data)class that explicitly encodes the return value's semantics be an option here? Another option would be to use a type alias that has a comment mentioning what each level corresponds to.